client/tso_client.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pd
+
+import (
+	"context"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/tsopb"
+)
+
+// TSOClient manages resource group info and token request.
+type TSOClient interface {
+	// GetTSWithinKeyspace gets a timestamp within the given keyspace from the TSO service
+	GetTSWithinKeyspace(ctx context.Context, keyspaceID uint32) (int64, int64, error)
```

Comment:
We also need other functions in this interface

Reply:
I only define Tso() rpc in protocolbuf and GetTSWithinKeyspaceAsync() will add later.

Reply:
Declare GetTSWithinKeyspaceAsync(), GetLocalTSWithinKeyspace() and GetLocalTSWithinKeyspaceAsync() in TSOClient.
    Remove the basic implementation of GetTSWithinKeyspace for now and will integrate TSOClient with PDClient's TSO batching/forwarding/async/pooling framework in the next few prs.